### ISSUE TITLE
Reference problem when using ABP-CLI

### ISSUE THREAD

**Issue by idemirel** [abpframework/abp#1408]
Hi,
When I use the CLI to create a template, i get a lot of reference problem.

Steps I do;

> D:\>**abp new QutupNet.AbpDemo -o D:\projects\dotnet\ID\training\QutupNet.AbpDemo**
> [17:39:22 INF] ABP CLI (https://abp.io)
> [17:39:23 INF] Version 0.18.1 (Stable channel)
> [17:39:26 INF] Installing yarn...
> C:\Users\idemirel\AppData\Roaming\npm\yarnpkg -> C:\Users\idemirel\AppData\Roaming\npm\node_modules\yarn\bin\yarn.js
> C:\Users\idemirel\AppData\Roaming\npm\yarn -> C:\Users\idemirel\AppData\Roaming\npm\node_modules\yarn\bin\yarn.js
> + yarn@1.16.0
> added 1 package in 2.548s
> [17:39:30 INF] Installing gulp...
> C:\Users\idemirel\AppData\Roaming\npm\gulp -> C:\Users\idemirel\AppData\Roaming\npm\node_modules\gulp\bin\gulp.js
> npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\gulp\node_modules\fsevents):
> npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
> 
> + gulp@4.0.2
> added 314 packages from 217 contributors in 24.463s
> [17:39:55 INF] Creating a new project...
> [17:39:55 INF] Project name: QutupNet.AbpDemo
> [17:39:56 INF] Using cached template: mvc, version: 0.18.1
> [17:39:56 INF] Check the documentation of this template: https://docs.abp.io/en/abp/latest/Startup-Templates/Mvc
> [17:39:58 INF] Successfully created the project 'QutupNet.AbpDemo'
> [17:39:58 INF] The output folder is: 'D:\projects\dotnet\ID\training\QutupNet.AbpDemo'

Then open project, i have lots of errors acording to references. One of project's csproj file is below (QutupNet.AbpDemo.Domain.Shared), all of them are project references which points missing directories instead of package reference (Nuget)

```
 <Project Sdk="Microsoft.NET.Sdk">

  <Import Project="..\..\common.props" />

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <RootNamespace>QutupNet.AbpDemo</RootNamespace>
  </PropertyGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\..\..\modules\identity\src\Volo.Abp.Identity.Domain.Shared\Volo.Abp.Identity.Domain.Shared.csproj" />
    <ProjectReference Include="..\..\..\..\modules\identityserver\src\Volo.Abp.IdentityServer.Domain.Shared\Volo.Abp.IdentityServer.Domain.Shared.csproj" />
    <ProjectReference Include="..\..\..\..\modules\background-jobs\src\Volo.Abp.BackgroundJobs.Domain.Shared\Volo.Abp.BackgroundJobs.Domain.Shared.csproj" />
    <ProjectReference Include="..\..\..\..\modules\audit-logging\src\Volo.Abp.AuditLogging.Domain.Shared\Volo.Abp.AuditLogging.Domain.Shared.csproj" />
    <ProjectReference Include="..\..\..\..\modules\tenant-management\src\Volo.Abp.TenantManagement.Domain.Shared\Volo.Abp.TenantManagement.Domain.Shared.csproj" />
    <ProjectReference Include="..\..\..\..\modules\feature-management\src\Volo.Abp.FeatureManagement.Domain.Shared\Volo.Abp.FeatureManagement.Domain.Shared.csproj" />
    <ProjectReference Include="..\..\..\..\modules\permission-management\src\Volo.Abp.PermissionManagement.Domain.Shared\Volo.Abp.PermissionManagement.Domain.Shared.csproj" />
    <ProjectReference Include="..\..\..\..\modules\setting-management\src\Volo.Abp.SettingManagement.Domain.Shared\Volo.Abp.SettingManagement.Domain.Shared.csproj" />
  </ItemGroup>

  <ItemGroup>
    <EmbeddedResource Include="Localization\AbpDemo\*.json" />
    <Content Remove="Localization\AbpDemo\*.json" />
  </ItemGroup>

</Project>
```

if I choose direct download using  [this link](https://abp.io/get-started) it works as expected.

thanks in advance

**Comment by tomlewandowski** [comment]
I have the same issue #1338

**Comment by maliming** [comment]
Possible cause: https://github.com/abpframework/abp/issues/1389

**Comment by hikalkan** [comment]
Will check this. We can release 0.18.2 hotfix if necessary.

**Comment by maliming** [comment]
@xuyae573 @tomlewandowski @idemirel

I temporarily compiled a CLI. [Click here to download the CLI](https://github.com/abpframework/abp/files/3343649/cli.zip)
If it solves your problem, we will release the 0.18.2 version of the CLI.

Download cli.zip, decompress and enter the cli directory, execute the relevant test command:
`dotnet .\Volo.Abp.Cli.dll new YourProject`

**Comment by idemirel** [comment]
@maliming 

it seems the same, i also clear the template cache.

![image](https://user-images.githubusercontent.com/156985/60409537-c27b3980-9bcc-11e9-8fec-04180d25d9a2.png)

result : 

![image](https://user-images.githubusercontent.com/156985/60409612-066e3e80-9bcd-11e9-8b53-5a48e9ea5005.png)


i also tried without "-o" switch, the same.

**Comment by maliming** [comment]
@idemirel Very strange, especially if your operating system is windows, this problem should not occur.

Do you have skype or other im software? We can check it remotely.

**Comment by Xeevis** [comment]
I'm unable to reproduce on Windows 10. This appears to be environmental (possibly cultural) issue meaning same code works differently in different environments. I took a rather blind shot at refactoring code that seems to be causing the problem. Can you please try [this precompiled CLI](https://github.com/Xeevis/abp/releases/download/1.18.2-alpha.1/Volo.Abp.Cli.zip).

Extract and from within the folder you can execute

    dotnet Volo.Abp.Cli.dll new QutupNet.AbpDemo -o D:\projects\dotnet\ID\training\QutupNet.AbpDemo

**Comment by idemirel** [comment]
i tried 0.19.0-dev , you built. it seems almost OK i think, the reference type should be "**PackageReference**" instead of "**ProjectReference**"i guess.
![image](https://user-images.githubusercontent.com/156985/60432310-9f6f7a80-9c0a-11e9-80c9-d0619b0841fb.png)

**Comment by Xeevis** [comment]
**Whoops, sorry about that**. But at least we fixed the issue yay! 😄 I've recompiled the updated code, this one should work for you (I've also changed version to a more correct 0.18.2-dev) you can use it until the changes are merged into main repo and proper 1.18.2 is released. [Download updated CLI](https://github.com/Xeevis/abp/releases/download/1.18.2-alpha.2/Volo.Abp.Cli.zip)

**Comment by idemirel** [comment]
That works now , thanks :)
